Moq.AutoMock/AutoMocker.cs

```diff
@@ -14,17 +14,39 @@ namespace Moq.AutoMock
     public partial class AutoMocker
     {
         private readonly Dictionary<Type, IInstance> typeMap = new Dictionary<Type, IInstance>();
-        private readonly MockBehavior mockBehavior;
+
+        public AutoMocker()
+        {
+            this.MockBehavior = MockBehavior.Default;
+            this.DefaultValue = DefaultValue.Empty;
+            this.CallBase = false;
+        }
 
         public AutoMocker(MockBehavior mockBehavior)
         {
-            this.mockBehavior = mockBehavior;
+            this.MockBehavior = mockBehavior;
+            this.DefaultValue = DefaultValue.Empty;
+            this.CallBase = false;
         }
 
-        public AutoMocker() : this(MockBehavior.Default)
+        public AutoMocker(MockBehavior mockBehavior, DefaultValue defaultValue)
         {
+            this.MockBehavior = mockBehavior;
+            this.DefaultValue = defaultValue;
+            this.CallBase = false;
         }
 
+        public AutoMocker(MockBehavior mockBehavior, DefaultValue defaultValue, bool callBase)
+        {
+            this.MockBehavior = mockBehavior;
+            this.DefaultValue = defaultValue;
+            this.CallBase = callBase;
+        }
+
+        public MockBehavior MockBehavior { get; set; }
+        public DefaultValue DefaultValue { get; set; }
+        public bool CallBase { get; set; }

```

Comment:
I would make these readonly properties. By leaving them mutable it opens up potentially undesired behavior if this `AutoMocker` instance creates some mocks, then these values are changes and other mocks are then created.

Reply:
Good call.  I'll make it so.